pennylane/optimize/riemannian_gradient.py

```diff
@@ -109,15 +109,13 @@ def algebra_commutator(tape, lie_algebra_basis_names, nqubits):
         qml.PauliRot(-np.pi / 2, pauli, wires=wires) for pauli in lie_algebra_basis_names
     )
 
-    measurements = [qml.expval(qml.s_prod(c, o)) for c, o in zip(*tape.measurements[0].obs.terms())]
-
-    tapes_plus = [QuantumScript(tape.operations + [op], measurements) for op in paulis_plus]
-    tapes_minus = [QuantumScript(tape.operations + [op], measurements) for op in paulis_minus]
+    tapes_plus = [tape.copy(operations=tape.operations + [op]) for op in paulis_plus]
+    tapes_minus = [tape.copy(operations=tape.operations + [op]) for op in paulis_minus]
```

Comment:
Could do
```suggestion
    new_tapes = [tape.copy(operations=tape.operations + [qml.PauliRot(sign * np.pi / 2, pauli, wires=wires)]) for sign in [1, -1] for pauli in lie_algebra_basis_names]
```
but maybe that's just less readable.

Reply:
Personally, I can forget about the order of multiple for loops in list comprehensions. I'd like to be a bit more explicit about first the plus tapes, then the minus tapes. But thanks for the recommendation :)